datafusion/sql/src/unparser/plan.rs

```diff
@@ -542,6 +553,11 @@ impl Unparser<'_> {
                     );
                 }
 
+                // Covers cases where the UNION is a subquery and the projection is at the top level
+                if select.already_projected() {
```

Comment:
Small improvement to correctly handle UNION unparsing for queries where UNION is a subquery